### DISCUSSION TITLE
DWeb Camp Workshop

### DISCUSSION THREAD

**Discussion by depatchedmode** [ChainAgnostic/secure-design#11] [Community Calls]
Here are my admittedly sparse notes from the DWeb Camp workshop. Thanks to all the attendees! Please fill the gaps on anything I missed while balancing the laptop on my knee 🫠.

![image](https://github.com/ChainAgnostic/secure-design/assets/84613835/c646332c-93dd-470d-b744-6baeb44be180)

- “Can we trust users to protect themselves?” 
	- Security perspective: Emphatic No, when assuming lowest common denominator data literacy.
- Should we scale friction automatically with the value of the assets?
- Safety vs Security vs Privacy - there’s tension. How do you knowingly make trade-offs?
- When you don’t know what the risk is with the interaction, systems are forced to assume the worst case scenario. Are there ways to make risk & tradeoffs more auditable?
- Some credit card vendors (eg. Target) have this notion of scaling friction with value. They will require extra factors and checks as purchase price increases.
- Big difference between credit card and crypto use cases: who bears the risk? In CC world, the issuer bears most risk. Who is it with crypto? Different answers for financial risk vs reputation risk?
- Is offline a concern as well? 
	- Yes. What is similar about the shape of this problem with offline-first systems versus systems that require online connectivity?
- There’s a mental model gap with how these systems work. 
	- Can we ever bridge that gap.
	- Users shouldn’t need to know the system in detail
	- Can we provide them with a minimal understanding of what to watch out for?
	- How do you help them properly comprehend the risks, and what happens with their digital footprint.
- Is anyone working on visual metaphors and design language? 
	- maybe this is skeuomorphism to or something else to indicate the flow
	- when you see a “spinning ball” you don’t know at what layer the bottleneck is occurring
	- often there is a lack of awareness amongst the practitioners 
	- Can we get a standard representation? There aren’t clear definitions of what needs standardization yet.

**Things to look at:**
- How To Ask Permission - https://github.com/ChainAgnostic/secure-design/discussions/4
- Yahoo Pattern Library - https://twitter.com/nathanacurtis/status/897157519151423489
- DecentPatterns.com - “decentralization off the shelf”
- Anoma - another layer 1
	- similar to “intents” - EVM
	- security: “what about side effects?” They are likely impossible to model.
- Delegatable.xyz
- [ReCap](https://blog.spruceid.com/extending-sign-in-with-ethereum-to-authorizations-recap/)

**Comment by depatchedmode** [comment]
Because Intents came up in this discussion, and may have been a new term for folks, here's a decent summary from a [video worth watching](https://www.youtube.com/watch?v=G0nFyq9DDPw):
<img width="1023" alt="image" src="https://github.com/ChainAgnostic/secure-design/assets/84613835/23ffe12d-721f-461f-948c-7cb0a48d90a4">

**Comment by agostbiro** [comment]
> Is anyone working on visual metaphors and design language?
> maybe this is skeuomorphism to or something else to indicate the flow
> when you see a “spinning ball” you don’t know at what layer the bottleneck is occurring
> often there is a lack of awareness amongst the practitioners
> Can we get a standard representation? There aren’t clear definitions of what needs standardization yet.

I've been thinking about solving this problem by tying user actions to Web2 counterparts. Not exactly skeuomorphism, but similar in principle. Maybe anachronism, lol.

Examples:

- sign a message to prove ownership of an address => sign in
- sign a message to attest to some fact (e.g. residence) => accept terms of service
- make a swap, mint an NFT, supply collateral for a loan, etc. => make a payment
- approve a token => this is a novel concept that we need to figure out how to convey, see https://github.com/ChainAgnostic/secure-design/discussions/9

I wrote more about this approach in this blog post: https://sealvault.org/blog/2023/05/ethereum-signature-approval/
